### PR TITLE
Fix for CR-1150166 & CR-1150159: add checks to handle out of bound heap access

### PR DESCRIPTION
**Pull request by saumyag-xilinx**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1150166
https://jira.xilinx.com/browse/CR-1150159
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
N/A
#### How problem was solved, alternative solutions (if any) and why they were rejected
The code was missing a call to the function that is performing necessary checks, so added the same.
#### Risks (if any) associated the changes in the commit
low
#### What has been tested and how, request additional testing if necessary
Tested using the Poc application provided in the CRs.
#### Documentation impact (if any)
N/A

### CODE REVIEW

File: src/runtime_src/core/common/drv/xrt_xclbin.c [hunk 1]
```diff
@@ -207,7 +207,11 @@ xrt_xclbin_get_section_hdr(const struct axlf *xclbin, enum axlf_section_kind kin
 		return NULL;
 
 	for (i = 0; i < xclbin->m_header.m_numSections; i++) {
-		if (xclbin->m_sections[i].m_sectionKind == kind)
+		if (xclbin->m_sections[i].m_sectionKind == kind){
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/common/drv/xrt_xclbin.c:210]: i dont see close braces . How did it compile?

**Comment by chvamshi-xilinx** [comment]
retest this please

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/common/drv/xrt_xclbin.c [hunk 2]
```diff
@@ -207,8 +207,13 @@ xrt_xclbin_get_section_hdr(const struct axlf *xclbin, enum axlf_section_kind kin
...
+			int err = 0;
+			err = xrt_xclbin_check_section_hdr(&(xclbin->m_sections[i]), xclbin->m_header.m_length);
```
Comment (stsoe) [inline src/runtime_src/core/common/drv/xrt_xclbin.c:212]: Is `xrt_xclbin_check_section_hdr` secure?  I think not, there can be overflow in the arithmetic such that the sum is less that `xclbin_len`, right?

I am reviewing the secure coding training we have been assigned.  And it has a good (not intuitive) example of how to deal with arithmetic overflow when checking for size.   

The check line 226 has to be strengthen to something like this:
`return ( (offset + size) < offset) || (offset + size) > xclbin_len) ? -EINVAL : 0;`

Comment (saumyag-xilinx) [reply src/runtime_src/core/common/drv/xrt_xclbin.c:212]: Thank you @stsoe, made suggested change.

Comment (mamin506) [inline src/runtime_src/core/common/drv/xrt_xclbin.c:212]: empty line in between declare and code.

File: src/runtime_src/core/common/drv/xrt_xclbin.c [hunk 3]
```diff
@@ -207,8 +207,13 @@ xrt_xclbin_get_section_hdr(const struct axlf *xclbin, enum axlf_section_kind kin
 		return NULL;
 
 	for (i = 0; i < xclbin->m_header.m_numSections; i++) {
-		if (xclbin->m_sections[i].m_sectionKind == kind)
+		if (xclbin->m_sections[i].m_sectionKind == kind){
```
Comment (mamin506) [inline src/runtime_src/core/common/drv/xrt_xclbin.c:210]:
```suggestion
		if (xclbin->m_sections[i].m_sectionKind == kind) {
```

File: src/runtime_src/core/common/drv/xrt_xclbin.c [hunk 4]
```diff
@@ -218,7 +223,7 @@ int
 xrt_xclbin_check_section_hdr(const struct axlf_section_header *header,
 	uint64_t xclbin_len)
 {
-	return (header->m_sectionOffset + header->m_sectionSize) > xclbin_len ?
+	return ((header->m_sectionOffset + header->m_sectionSize < header->m_sectionOffset) || (header->m_sectionOffset + header->m_sectionSize > xclbin_len)) ?
```
Comment (mamin506) [inline src/runtime_src/core/common/drv/xrt_xclbin.c:226]: Can we break this into two lines for better readability？

```suggestion
	return ((header->m_sectionOffset + header->m_sectionSize < header->m_sectionOffset)
	            || (header->m_sectionOffset + header->m_sectionSize > xclbin_len)) ?
```

**Review by mamin506 — CHANGES_REQUESTED** [review]

**Review by mamin506 — APPROVED** [review]
I have few coding style change. The logic looks good. I will approve this for 2023.1 release.

I will file a CR for refine the coding style in 2023.2,

**Pull request merged**